src/pixie/context.nim

```diff
@@ -113,17 +118,31 @@ proc restore*(ctx: Context) =
     else: # Otherwise draw to the root image
       ctx.image.draw(poppedLayer)
 
-proc fill(
-  ctx: Context, image: Image, path: Path, windingRule: WindingRule
-) {.inline.} =
+proc fill(ctx: Context, image: Image, path: Path, windingRule: WindingRule) =
+  var image = image
+
+  if ctx.globalAlpha != 1:
+    ctx.saveLayer()
```

Comment:
Are you sure we can't just changes alpha on fill?

Reply:
This is done the same way as cpurender2. Paint could be an image or a gradient or who knows what. There's no good way that I know of to handle all of that with an alpha applied at fill time that we've implemented afaik.